src/openrct2/Context.cpp

```diff
@@ -713,6 +721,12 @@ namespace OpenRCT2
 
         void CloseProgress() override
         {
+            const auto isMainThread = _mainThreadId == std::this_thread::get_id();
+            if (!gOpenRCT2Headless && !isMainThread)
```

Comment:
These checks differs from the existing one on line -705/+713, which only executes the code if we are on the main thread and if we are not headless.

Which leads me to believe this check should read
```suggestion
            if (gOpenRCT2Headless || !isMainThread)
```
I would be interested to hear if I got that correct or not.